### PR TITLE
Implement timeout using react-countdown package and timestamp instead of intervals and replace some usestate variables to prevent multiple rerendering

### PR DESCRIPTION
**Pull request by Avelous**

### CODE REVIEW

File: packages/nextjs/app/game/[id]/page.tsx [hunk 1]
```diff
@@ -19,111 +19,166 @@ import { notification } from "~~/utils/scaffold-eth";
...
+  const isUpdatingRoundRef = useRef(isUpdatingRound);
+  const [updateRoundCountdown, setCountdown] = useState(20);
+  const [showRoundCountdown, setshowRoundCountdown] = useState(false);
+  const [nextRoundCountdown, setNextRoundCountdown] = useState(5);
   const [pauseAtRoundsEnd, setPauseAtRoundsEnd] = useState(false);
-  const [timeout, setTimeOut] = useState(60);
-  // const [isTimingout, setIsTimingout] = useState(false);
+  const pauseAtRoundsEndRef = useRef(pauseAtRoundsEnd);
+
+  const [isTimingout, setIsTimingout] = useState(false);
```
Comment (damianmarti) [inline packages/nextjs/app/game/[id]/page.tsx]: Maybe change to isEnding

File: packages/nextjs/app/game/[id]/page.tsx [hunk 2]
```diff
@@ -19,111 +19,166 @@ import { notification } from "~~/utils/scaffold-eth";
...
-  const [player, setPlayer] = useState<playerType>();
+  const gameRef = useRef<Game | null>(null);
+  const playerRef = useRef<playerType | null>(null);
   const [token, setToken] = useState("");
   const [isUpdatingRound, setIsUpdatingRound] = useState(false);
-  const [countdown, setCountdown] = useState(20);
-  const [showCountdown, setShowCountdown] = useState(false);
+  const isUpdatingRoundRef = useRef(isUpdatingRound);
+  const [updateRoundCountdown, setCountdown] = useState(20);
+  const [showRoundCountdown, setshowRoundCountdown] = useState(false);
```
Comment (damianmarti) [inline packages/nextjs/app/game/[id]/page.tsx]: setShowRoundCountdown

File: packages/nextjs/app/game/[id]/page.tsx [hunk 3]
```diff
@@ -19,111 +19,166 @@ import { notification } from "~~/utils/scaffold-eth";
...
-  const [game, setGame] = useState<Game>();
-  const [player, setPlayer] = useState<playerType>();
+  const gameRef = useRef<Game | null>(null);
+  const playerRef = useRef<playerType | null>(null);
   const [token, setToken] = useState("");
   const [isUpdatingRound, setIsUpdatingRound] = useState(false);
-  const [countdown, setCountdown] = useState(20);
-  const [showCountdown, setShowCountdown] = useState(false);
+  const isUpdatingRoundRef = useRef(isUpdatingRound);
+  const [updateRoundCountdown, setCountdown] = useState(20);
```
Comment (damianmarti) [inline packages/nextjs/app/game/[id]/page.tsx]: fix names

**Review by damianmarti — APPROVED** [review]
Some small comments but everything seems fine! Great work!

File: packages/nextjs/app/game/[id]/page.tsx (hunk 3 above)
Comment (Avelous) [reply packages/nextjs/app/game/[id]/page.tsx]: Fixed!!

File: packages/nextjs/app/game/[id]/page.tsx (hunk 2 above)
Comment (Avelous) [reply packages/nextjs/app/game/[id]/page.tsx]: Thanks for spotting these

File: packages/nextjs/app/game/[id]/page.tsx (hunk 1 above)
Comment (Avelous) [reply packages/nextjs/app/game/[id]/page.tsx]: Replaced with isRoundEnding

**Pull request merged**